### PR TITLE
Portal Gallery and Submission Details

### PR DESCRIPTION
**Pull request by he-is-james**
- adjusted Submission Detail Screen
- edit button allows admin to edit title and description of a Submission, cancel will not update and save will update the database
- submit only available if the submission is incomplete and will update the status and comments
- change will be similar to edit but only update status and comments

### CODE REVIEW

File: mobile/Screen/GalleryScreen.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ function GalleryScreen({ navigation }) {
               allImageData.map((imageData) => (
                 <ArtworkCard
                   ImageURL={imageData.ImageURL}
-                  id={imageData.SubmissionData._id}
+                  id={imageData.SubmissionId}
```
Comment (siddharthnandy) [inline mobile/Screen/GalleryScreen.js:57]: will be changed with convention change in submissionController

File: website/src/Pages/GalleryPage/GalleryPage.jsx [hunk 2]
```diff
@@ -1,42 +1,60 @@
...
-      'Submission Title': 'Taco Tuesday',
-      Uploader: 'Isaac Wen',
-      Status: 'Incomplete',
-      Category: 'Image',
-      Date: '02/01/23',
-    },
-  ];
+
+  const getSubmissionData = async () => {
+    const response = await axios.get('http://localhost:4000/submissions/getsubmissions');
```
Comment (siddharthnandy) [inline website/src/Pages/GalleryPage/GalleryPage.jsx:20]: Maybe keep the URL in a .env file like how we work with requests on mobile?

File: website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+  const handleEditCancel = () => {
+    setTitle(details.title);
+    setDescription(details.description);
+    setEdit(false);
+  };
+
+  const handleEditSave = async () => {
+    // TODO: pop up message for the approval or switching of status
+    const response = await axios.patch(`http://localhost:4000/submissions/updatesubmission/${id}`, {
```
Comment (siddharthnandy) [inline website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx:74]: env URL?

File: website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  const [edit, setEdit] = useState(false);
+  const [change, setChange] = useState(false);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [action, setAction] = useState('');
+  const [comments, setComments] = useState('');
+  const [update, setUpdate] = useState(0);
+
+  const getSubmissionDetails = async () => {
+    const artwork = await axios.get('http://localhost:4000/submissions/getartwork', { params: { id } });
```
Comment (siddharthnandy) [inline website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx:34]: env URL?

File: website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+      setAction('Approve');
+    } else {
+      setAction('Reject');
+    }
+    setComments(artwork.data.Submission.comments);
+  };
+
+  useEffect(() => {
+    getSubmissionDetails();
+  }, [update]);
```
Comment (siddharthnandy) [inline website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx:62]: is re-retrieving the entire Submission anytime admin updates a good idea, or can we not just work with whatever they inputted when they updated the submission?

File: website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+    setUpdate((val) => val + 1);
+    setEdit(false);
+  };
+
+  const handleSubmit = async () => {
+    let formatStatus = 'Approved';
+    if (action === 'Reject') {
+      formatStatus = 'Rejected';
+    }
+    const response = await axios.patch(`http://localhost:4000/submissions/updatesubmission/${id}`, {
```
Comment (siddharthnandy) [inline website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx:88]: env URL?

File: website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+    setChange(false);
+  };
+
+  const handleChangeSave = async () => {
+    // TODO: pop up message for the approval or switching of status
+    let formatStatus = 'Approved';
+    if (action === 'Reject') {
+      formatStatus = 'Rejected';
+    }
+    const response = await axios.patch(`http://localhost:4000/submissions/updatesubmission/${id}`, {
```
Comment (siddharthnandy) [inline website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx:108]: env URL?

File: website/src/Pages/ZinesPage/ZineCard.jsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+}) {
+  const dispatch = useDispatch();
+
+  // TODO: create mini pop up
+  const handleEditZineDetails = async () => {
+    console.log(id);
+  };
+
+  const handleDeleteZineDetails = async () => {
+    await axios.delete(`http://localhost:4000/zine/delete/${id}`);
```
Comment (siddharthnandy) [inline website/src/Pages/ZinesPage/ZineCard.jsx:21]: env URL?

File: website/src/Pages/ZinesPage/ZinesPage.jsx [hunk 9]
```diff
@@ -1,10 +1,38 @@
...
+import { Box, Container, Typography } from '@mui/material';
+import axios from 'axios';
+import ZineCard from './ZineCard';
 
 function ZinesPage() {
+  const { count } = useSelector((state) => state.sliceAuth);
+  const [zines, setZines] = useState([]);
+
+  const getZines = async () => {
+    const response = await axios.get('http://localhost:4000/zine/get');
```
Comment (siddharthnandy) [inline website/src/Pages/ZinesPage/ZinesPage.jsx:12]: env URL?

**Review by siddharthnandy — CHANGES_REQUESTED** [review]

File: website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx (hunk 6 above)
Comment (he-is-james) [reply website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx:88]: clean up

**Review by siddharthnandy — APPROVED** [review]

**Pull request merged**